src/HonzaBotner.Discord.Services/Commands/PinCommands.cs

```diff
@@ -129,21 +125,16 @@ public async Task DeleteInChannel(CommandContext ctx, DiscordChannel channel)
                 foreach (DiscordMessage message in messages)
                 {
                     int score = 0;
-                    IReadOnlyList<DiscordUser> reactions = await message.GetReactionsAsync(permanentEmoji, 10);
+                    IReadOnlyList<DiscordUser> reactions = await message.GetReactionsAsync(permanentEmoji, _pinOptions.Treshold);
 
-                    if (reactions.Count == 10) continue;
+                    if (reactions.Count == _pinOptions.Treshold) continue;
 
                     foreach (DiscordUser user in reactions)
                     {
                         int maxRoleScore = 1;
                         DiscordMember member = await channel.Guild.GetMemberAsync(user.Id);
```

Comment:
Ergo we should also iterate through all the reactions.

Reply:
I don't understand what can fail. We are comparing two numbers and skipping the count in case they match. Anything else skips this count and iterates through reactins manually as you propose.